include/envoy/filesystem/filesystem.h

```diff
@@ -61,5 +62,48 @@ class Watcher {
 
 typedef std::unique_ptr<Watcher> WatcherPtr;
 
+enum class FileType { Regular, Directory, Other };
+
+struct DirectoryEntry {
+  // name_ is the name of the file in the directory, not including the directory path itself
+  // For example, if we have directory a/b containing file c, name_ will be c
+  std::string name_;
+
+  // Note that if the file represented by name_ is a symlink, type_ will be the file type of the
+  // target. For example, if name_ is a symlink to a directory, its file type will be Directory.
+  FileType type_;
+
+  bool operator==(const DirectoryEntry& rhs) const {
+    return name_ == rhs.name_ && type_ == rhs.type_;
+  }
+};
+
+/**
+ * Abstraction for listing a directory.
+ * TODO(sesmith177): replace with std::filesystem::directory_iterator once we move to C++17
+ */
+class DirectoryIteratorImpl;
+class DirectoryIterator {
+public:
+  typedef std::input_iterator_tag iterator_category;
+  typedef DirectoryEntry value_type;
+  typedef DirectoryEntry& reference;
+  typedef void pointer;
+  typedef void difference_type;
```

Comment:
Actually, what are these all for?

Reply:
These are types that need to be defined for a class to be an iterator (we found this information here: https://en.cppreference.com/w/cpp/iterator/iterator)

Since we only use this class with range-based for loops, which don't require an iterator (just overloaded operators *, ++ and !=), we can probably just delete the typedefs